### PR TITLE
ISSUE #1937: Remove request number

### LINKED ISSUE #1937: Request # on the Domain Request Detail page is confusing
### Issue description

Problem: Requesters who have submitted a request we'll see the "Request #" near the top of the detail page. As that number grows, this may cause confusion on a part of the requester, feeling that they are deep in a queue awaiting processing. This may lead to more inquiries on the helpdesk.

Solution: since this number is no value for the sake of look up, and this for reference only, we feel it best to simply remove this from the domain request detail page view.

### Acceptance criteria

- [ ] The "Request #" no longer appears on the request detail page.

### Additional context

See the attached image, and note the position of "Request #" This reference should be removed.

Image

### Links to other issues

_No response_

### PR DESCRIPTION
**Pull request by therealslimhsiehdy**
## Ticket

Resolves #1937 

## Changes

* Remove request number from the Request Domain Detail page

## Context for reviewers

The number can confusion to the requester and feel that they are deep in a queue awaiting processing. This may lead to more inquiries on the helpdesk so we want to remove it to avoid confusion.

## Setup

1. Go to the [RH sandbox](https://getgov-rh.app.cloud.gov/)
3. Go to a domain that has been submitted and see that the request number is removed. See screenshot for reference

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

## Screenshots
Screenshot 2024-03-29 at 9 25 21 AM

### CODE REVIEW

**Comment by erinysong** [comment]
Verifying that request number no longer is displayed as expected.
**rh-sandbox** - request number removed
image

**other domain on es-sandbox** - request number still exists without this PR
image

**Review by erinysong — APPROVED** [review]
LGTM! Tested and verified UI changes work as expected 🎉

File: src/registrar/templates/domain_request_status.html [hunk 1]
```diff
@@ -41,8 +41,7 @@ <h1>Domain request for {{ DomainRequest.requested_domain.name }}</h1>
       </div>
     </div>
     <br>
-    <p> <b class="review__step__name">Last updated:</b> {{DomainRequest.updated_at|date:"F j, Y"}}<br>
-    <b class="review__step__name">Request #:</b> {{DomainRequest.id}}</p>
+    <p> <b class="review__step__name">Last updated:</b> {{DomainRequest.updated_at|date:"F j, Y"}}<br></p>
```
Comment (rachidatecs) [inline src/registrar/templates/domain_request_status.html:44]: `<p><b class="review__step__name">Last updated:</b> {{DomainRequest.updated_at|date:"F j, Y"}}</p>`

You should no longer need that br. Test and make sure.

**Review by rachidatecs — COMMENTED** [review]
You know what, I'm actually gonna request changes :/

**Pull request merged**